### PR TITLE
fix mouse wheel can cause the phone number to negative

### PR DESCRIPTION
**Pull request by wanghaha1991**
Re: cocos-creator/3d-tasks#

Changelog:
 * fix mouse wheel can cause the phone number to negative

### CODE REVIEW

File: cocos2d/core/components/editbox/WebEditBoxImpl.js [hunk 1]
```diff
@@ -434,6 +434,7 @@ Object.assign(WebEditBoxImpl.prototype, {
         } else if(inputMode === InputMode.PHONE_NUMBER) {
             type = 'number';
             elem.pattern = '[0-9]*';
+            elem.onmousewheel = 'return false;';
```
Comment (PPpro) [inline cocos2d/core/components/editbox/WebEditBoxImpl.js]:
```suggestion
            elem.onmousewheel = function () { return false; };
```

Comment (PPpro) [reply cocos2d/core/components/editbox/WebEditBoxImpl.js]: Thanks for feedback
we thought that the eval string is not a good practice, and it doesn't work well on the Chrome on MacOS

maybe it is better to change like this.

Comment (wanghaha1991) [reply cocos2d/core/components/editbox/WebEditBoxImpl.js]: I have no local compilation environment, so I directly get the input object from the implementation of Cocos Creator 2.4.2:
I'll create a new Editbox on a blank scene and test it as follows:
```
   onLoad: function () {
        this.editBox._impl._elem.setAttribute("onmousewheel","function () { return false; };");
        console.log( this.editBox._impl._elem);
    },
```
chrome report error:
image


Using native HTML5 tests, error as above (missing function name):
```
 <form>
        name: <input type="number" name="number" onmousewheel="function a(){ return false; };"><br>
</form>
```
After add a function name：
```
<form>
        name: <input type="number" name="number" onmousewheel="function a(){ return false; };"><br>
 </form>

 this.editBox._impl._elem.setAttribute("onmousewheel","function a() { return false; };");
```
The mouse wheel effect does not fail

Comment (PPpro) [reply cocos2d/core/components/editbox/WebEditBoxImpl.js]: ## the difference is that
```
elem.onmousewheel = "function () { return false; };"
```
means we **declare** a nameless function at the mousewheel callback, which has never been called


```
elem.onmousewheel = function () { return false; };
```
means we **call** the namelesss function  at the mousewheel callback, it's kind of like:
```
elem.onmousewheel = "(function () { return false; })();"
```

Comment (wanghaha1991) [reply cocos2d/core/components/editbox/WebEditBoxImpl.js]: Thank you for your answer. I learned a lot. I am a back-end developer，I just started working on cocos. 😸
I have updated this PR

**Review by PPpro — APPROVED** [review]

**Comment by SantyWang** [comment]
Thanks for your contribution again

**Pull request merged**